### PR TITLE
PSR-569 Why does scheme hold shares page

### PR DESCRIPTION
**Pull request by kberec**

### CODE REVIEW

File: conf/shares.routes [hunk 1]
```diff
@@ -11,7 +11,12 @@ POST        /:srn/change-total-value-quoted-shares
...
+GET         /:srn/change-which-shares-held-transaction/:index                           controllers.nonsipp.shares.TypeOfSharesHeldController.onPageLoad(srn: Srn, index: Max5000, mode: Mode = CheckMode)
+POST        /:srn/change-which-shares-held-transaction/:index                           controllers.nonsipp.shares.TypeOfSharesHeldController.onSubmit(srn: Srn, index: Max5000, mode: Mode = CheckMode)
+
+GET         /:srn/why-does-scheme-hold-shares/:index                                  controllers.nonsipp.shares.WhyDoesSchemeHoldSharesController.onPageLoad(srn: Srn, index: Max5000, mode: Mode = NormalMode)
+POST        /:srn/why-does-scheme-hold-shares/:index                                  controllers.nonsipp.shares.WhyDoesSchemeHoldSharesController.onSubmit(srn: Srn, index: Max5000, mode: Mode = NormalMode)
+GET         /:srn/change-why-does-scheme-hold-sharesn/:index                           controllers.nonsipp.shares.WhyDoesSchemeHoldSharesController.onPageLoad(srn: Srn, index: Max5000, mode: Mode = CheckMode)
```
Comment (ccatosdevelopment) [inline conf/shares.routes:21]: Looks like a rogue character (n) snuck into the url here

Comment (kberec) [reply conf/shares.routes:21]: Good spot @ccatosdevelopment , we are not using change- URLs per now, so I'll change it with the next ticket.

File: test/controllers/nonsipp/shares/WhyDoesSchemeHoldSharesControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    act.like(renderView(onPageLoad, userAnswers) { implicit app => implicit request =>
+      val view = injected[RadioListView]
+
+      view(
+        form(injected[RadioListFormProvider]),
+        viewModel(srn, index, schemeName, TypeOfShares.ConnectedParty.name, NormalMode)
+      )
+    })
+
```
Comment (ccatosdevelopment) [inline test/controllers/nonsipp/shares/WhyDoesSchemeHoldSharesControllerSpec.scala:51]: These tests look good, but I've noticed that similar pages on other journeys also have a 'renderPrePopView' test (e.g. lines 48-61 in [WhyDoesSchemeHoldLandPropertyControllerSpec](https://github.com/hmrc/pension-scheme-return-frontend/blob/5978dc8f6924e484dbbda436a6a6ff84542200b8/test/controllers/nonsipp/landorproperty/WhyDoesSchemeHoldLandPropertyControllerSpec.scala#L48)) - is a similar test required in this case or not?

Comment (kberec) [reply test/controllers/nonsipp/shares/WhyDoesSchemeHoldSharesControllerSpec.scala:51]: I can have a look. Is it ok to do that in the next branch, so to get this merge and unblock other tickets?

Comment (ccatosdevelopment) [reply test/controllers/nonsipp/shares/WhyDoesSchemeHoldSharesControllerSpec.scala:51]: Yup, absolutely!

**Review by ccatosdevelopment — APPROVED** [review]

**Pull request merged**